tracing/src/main/java/com/palantir/tracing/Tracers.java

```diff
@@ -174,6 +179,45 @@ public static Runnable wrap(String operation, Runnable delegate) {
         return new TracingAwareRunnable(Optional.of(operation), delegate);
     }
 
+    /**
+     * Traces the the execution of the <pre>delegateFactory</pre>, completing a span when the returned
+     * {@link ListenableFuture#isDone() ListenableFuture is done}.
+     *
+     * Example usage:
+     * <pre>{@code
+     *     ListenableFuture<Result> future = Tracers.wrapListenableFuture(
+     *         "remote operation",
+     *         () -> retrofitClient.doRequest());
+     * }</pre>
+     *
+     * Note that this function is not named <pre>wrap</pre> in order to avoid conflicting with potential utility
+     * methods for {@link Supplier suppliers}.
+     */
+    @Beta
+    public static <T, U extends ListenableFuture<T>> U wrapListenableFuture(
+            String operation, Supplier<U> delegateFactory) {
+        DetachedSpan span = DetachedSpan.start(operation);
+        U result = null;
+        // n.b. This span is required to apply tracing thread state to an initial request. Otherwise if there is
+        // no active trace, the detached span would not be associated with work initiated by delegateFactory.
+        try (CloseableSpan ignored =
+                // This could be more efficient using https://github.com/palantir/tracing-java/issues/177
+                span.childSpan("Initial: " + operation)) {
```

Comment:
I think I'd like to keep the user-supplied operation name at the beginning so that it's easier to scan in UIs (especially when the span name appears in a column which might be too narrow).

Thoughts on just appending ` (initial)` or ` (ListenableFuture)` ?

Reply:
sounds reasonable to me.